### ISSUE TITLE
NETLINK_ROUTE support

### ISSUE THREAD

**Issue by ianlewis** [google/gvisor#578]
Much of netlink socket route support is not implemented. Need to figure out the full scope of work needed to fully support.
See: http://man7.org/linux/man-pages/man7/rtnetlink.7.html

**Comment by prattmic** [comment]
The existing support is sufficient for glibc getifaddrs(3) and basic ip route/addr and ifconfig support.

**Comment by ianlewis** [comment]
ip route is a WIP and will get merged shortly. ip addr and ifconfig work right now.
gvisor.dev/pr/508

**Comment by liornm** [comment]
It seems like "ip route add" (RTM_NEWROUTE) is not yet supported (despite the check mark).

`else if hdr.Flags&linux.NLM_F_REQUEST == linux.NLM_F_REQUEST {
		switch hdr.Type {
		case linux.RTM_GETLINK:
			return p.getLink(ctx, msg, ms)
		case linux.RTM_GETROUTE:
			return p.dumpRoutes(ctx, msg, ms)
		case linux.RTM_NEWADDR:
			return p.newAddr(ctx, msg, ms)
		case linux.RTM_DELADDR:
			return p.delAddr(ctx, msg, ms)
		default:
			return syserr.ErrNotSupported
		}
	}`
Am I missing something?

**Comment by ianlewis** [comment]
@liornm Yes. I'm not sure why that was checked. I updated the checklist.

**Comment by ianlewis** [comment]
@liornm No one is currently working on it. Feel free to submit a PR for the parts you need.

**Comment by liornm** [comment]
Does the `stack.RouteTable()` ever being used for network routing purposes? Because I don't see any references except for the NETLINK_ROUTE.
Otherwise, where is the implementation of the actual routing table?

**Comment by ianlewis** [comment]
RouteTable is only used for returning the route table to the user via netlink right now. If you're looking at adding RTM_NEWROUTE support, you'd likely need to add a new method to the [inet.Stack interface](https://github.com/google/gvisor/blob/master/pkg/sentry/inet/inet.go#L24-L101) and then implement where necessary. I think that's just [netstack](https://github.com/google/gvisor/blob/master/pkg/sentry/socket/netstack/stack.go) and [hostinet](https://github.com/google/gvisor/blob/master/pkg/sentry/socket/hostinet/stack.go).
